### PR TITLE
fix(node): Enable e2e test

### PR DESCRIPTION
**Pull request by ndcunningham**
Re-enables e2e tests for node 

Docker changes to be merged first: https://github.com/nrwl/nx/pull/26365

### CODE REVIEW

File: e2e/node/src/node-server.test.ts [hunk 1]
```diff
@@ -14,11 +14,16 @@ import {
 } from '@nx/e2e/utils';
 import { join } from 'path';
 
-xdescribe('Node Applications + webpack', () => {
+describe('Node Applications + webpack', () => {
```
Comment (FrozenPandaz) [inline e2e/node/src/node-server.test.ts:17]: There's another `xdescribe` in this file. Could you enable that as well please?

File: packages/node/src/generators/setup-docker/setup-docker.ts [hunk 2]
```diff
@@ -30,14 +28,16 @@ function normalizeOptions(
...
 
   if (
     !projectConfig ||
     !projectConfig.targets ||
     !projectConfig.targets[options.buildTarget]
   ) {
+    logger.error(
+      `Could not find the project ${options.project} or the build target ${options.buildTarget} in the workspace.`
+    );
     return;
```
Comment (FrozenPandaz) [inline packages/node/src/generators/setup-docker/setup-docker.ts:41]: Should this throw an error?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: e2e/node/src/node-server.test.ts (hunk 1 above)
Comment (ndcunningham) [reply e2e/node/src/node-server.test.ts:17]: there are no `xdescribe`s in this file you can see the output here: https://staging.nx.app/runs/oaKPGjBQVJ?utm_source=pull-request&utm_medium=comment

File: packages/node/src/generators/setup-docker/setup-docker.ts (hunk 2 above)
Comment (ndcunningham) [reply packages/node/src/generators/setup-docker/setup-docker.ts]: Updated

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/node/src/generators/setup-docker/setup-docker.ts [hunk 3]
```diff
@@ -30,15 +28,16 @@ function normalizeOptions(
 }
 
 function addDocker(tree: Tree, options: SetUpDockerOptions) {
-  // Inferred targets are only available in the project graph
-  const projectConfig = readCachedProjectConfiguration(options.project);
+  const projectConfig = readProjectConfiguration(tree, options.project);
```
[the remark this answers is not in the document]
Comment (ndcunningham) [reply packages/node/src/generators/setup-docker/setup-docker.ts:31]: Why not?

1. The project may not exist on file as yet so reading only from the graph would be invalid.
2. The `tree` is passed in as an argument, which is probably the source of truth for generators.

The current setup should work for both crystal and non-inferred projects.

**Review by jaysoo — APPROVED** [review]

File: e2e/node/src/node-server.test.ts [hunk 4]
```diff
@@ -14,11 +14,16 @@ import {
...
   let proj: string;
   beforeAll(() => {
     proj = newProject({
       packages: ['@nx/node'],
+      // npm has resolution for ajv some packages require ajv6 and some require ajv8 and npm resolves it to ajv6 (Error: Cannot find module 'ajv/dist/compile/codegen')
```
Comment (FrozenPandaz) [inline e2e/node/src/node-server.test.ts:22]: We should at least add a //TODO to re-enable the tests for other package managers.

@jaysoo Do you know how we can either align these versions or mitigate this issue with the default options of npm?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**